### PR TITLE
e2e tests to ensure no broken links

### LINKED ISSUE #2228: Ensure e2es would catch broken local links
**Is your feature request related to a problem? Please describe.**
E2e tests should check every link in every environment, and ensure we never get a 404 NB only check they're not 404, as we don't want to test if the pages we're pointing at are erroring (5xx) or not (2xx).

**Describe the solution you'd like**
Actually, test this, because we broke them all in https://github.com/bbc/simorgh/pull/2187 and tests still passed

Maybe iterate over all links and do a cy.request (quicker than clicking on them and coming back)

**Describe alternatives you've considered**
n/a

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### PR DESCRIPTION
**Pull request by PriyaKR**
Resolves #2228

**Overall change:** _All the links on the page are asserted for non 404 status code_

**Code changes:**

- _Added tests inside `index.js` to ensure all the page links are not broken_

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by PriyaKR** [comment]
I have added tests to verify footer links when it comes to header some services seems to have top navigation where the links takes to error pages.Also the banner branding is covered by the existing tests.

File: cypress/integration/pages/index.js [hunk 1]
```diff
@@ -96,6 +96,15 @@ const runCommonTests = ({ service, pageType }) => {
         );
       });
 
+      it('footer links should not 404', () => {
+        cy.get(' footer a').each(element => {
```
Comment (dr3) [inline cypress/integration/pages/index.js]:
```suggestion
        cy.get('footer a').each(element => {
```

**Review by thekp — APPROVED** [review]

**Review by oluoluoxenfree — APPROVED** [review]

**Review by dr3 — APPROVED** [review]

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]
If this is only broken on anchor links then we can run on all envs, but we need to ensure it passes on anchor links - or exclude them in the cy.get(), if this is possible.

File: cypress/integration/pages/index.js [hunk 2]
```diff
@@ -96,6 +96,15 @@ const runCommonTests = ({ service, pageType }) => {
         );
       });
 
+      it('footer links should not 404', () => {
```
Comment (dr3) [inline cypress/integration/pages/index.js]:
```suggestion
      it('links should not 404', () => {
```

**Comment by PriyaKR** [comment]
@benjaminhobbs i tried excluding the anchor links using `cy.get('a')
          .not('[href="#content"]')....` which excluded the anchor links but it appears to throw 404  on some of the nav links
for eg:
2019-08-21 12:46:30 info [server/index.jsx] GET /igbo/egwuregwu 404 153 - 0.593 ms
        2) footer links should not 404
So i am going to try the other option of running the tests only for live env

File: cypress/integration/pages/index.js [hunk 3]
```diff
@@ -96,6 +96,19 @@ const runCommonTests = ({ service, pageType }) => {
         );
       });
 
+      if (Cypress.env('APP_ENV') === 'live') {
+        it('footer links should not 404', () => {
+          cy.get('a')
+            .not('[href="#content"]')
```
Comment (jamesbhobbs) [inline cypress/integration/pages/index.js]: This seems a little too specific, but a .not (starting with #) would be ideal I think.

**Review by jamesbhobbs — APPROVED** [review]

**Pull request merged**